### PR TITLE
[Chat][FluentUI] Resolve Issues with compact mode

### PR DESCRIPTION
**Pull request by palatter**
# What

# Why

# How Tested

# Process & policy checklist

- [ ] I have updated the project documentation to reflect my changes if necessary.
- [ ] I have read the [CONTRIBUTING](https://github.com/Azure/communication-ui-library/blob/main/CONTRIBUTING.md) documentation.

**Is this a breaking change?**

- [ ] This change causes current functionality to break.

### CODE REVIEW

File: packages/react-components/src/components/ChatMessage/ChatMessageComponentAsMessageBubble.tsx [hunk 1]
```diff
@@ -35,7 +35,9 @@ import { mergeClasses } from '@fluentui/react-components';
 import {
   chatBlockedMessageClasses,
   chatBlockedMyMessageClasses,
-  defaultChatItemMessageContainerNoOverlap,
+  // defaultChatItemMessageContainerNoOverlap,
```
Comment (vhuseinova-msft) [inline packages/react-components/src/components/ChatMessage/ChatMessageComponentAsMessageBubble.tsx:38]: Is commented code still needed?

File: packages/react-components/src/components/styles/MessageThread.styles.ts [hunk 2]
```diff
@@ -113,11 +113,24 @@ const defaultChatItemMessageContainer = (overlapAvatarAndMessage: boolean): Comp
...
-/**
- * @private
- */
-export const defaultChatItemMessageContainerOverlap = makeStyles(defaultChatItemMessageContainer(true));
+export const defaultChatItemMessageContainerStyles = makeStyles({
+  body: {
+    marginRight: '0rem',
+    zIndex: CHAT_MESSAGE_ZINDEX,
+    '& msft-mention': {
+      color: '#D83B01',
```
Comment (vhuseinova-msft) [inline packages/react-components/src/components/styles/MessageThread.styles.ts:121]: We should use theme values for colors (tokens if possible or something like was done in `const errorTextColor = 'var(--errorText)';`)

File: packages/react-components/src/components/styles/MessageThread.styles.ts [hunk 3]
```diff
@@ -113,11 +113,24 @@ const defaultChatItemMessageContainer = (overlapAvatarAndMessage: boolean): Comp
...
+      fontWeight: 600
```
Comment (vhuseinova-msft) [inline packages/react-components/src/components/styles/MessageThread.styles.ts:122]: same for the font weight if possible

File: packages/react-components/src/components/ChatMessage/ChatMessageComponentAsMessageBubble.tsx (hunk 1 above)
Comment (palatter) [reply packages/react-components/src/components/ChatMessage/ChatMessageComponentAsMessageBubble.tsx:38]: Good call! Will remove.

File: packages/react-components/src/components/ChatMessage/ChatMessageComponentAsMessageBubble.tsx [hunk 4]
```diff
@@ -303,6 +308,15 @@ const MessageBubble = (props: ChatMessageComponentAsMessageBubbleProps): JSX.Ele
   const myMessageContainerClasses = isBlockedMessage ? chatBlockedMyMessageClass : chatMyMessageClass;
   const messageContainerClasses = isBlockedMessage ? chatBlockedMessageClass : chatMessageClass;
   const rootLayout = _useChatMyMessageLayout();
+
+  const chatItemMessageContainerClasses = defaultChatItemMessageContainerStyles();
+  const chatItemMessageContainerClassName = mergeClasses(
+    // Need to merge the styles injected with props - Future PR
```
Comment (emlynmac) [inline packages/react-components/src/components/ChatMessage/ChatMessageComponentAsMessageBubble.tsx:314]: Is there a task for this change?

**Review by emlynmac — APPROVED** [review]

File: packages/react-components/src/components/ChatMessage/ChatMessageComponentAsMessageBubble.tsx (hunk 4 above)
Comment (palatter) [reply packages/react-components/src/components/ChatMessage/ChatMessageComponentAsMessageBubble.tsx:314]: I belive so. Just linked it to you in teams.

**Pull request merged**